### ISSUE TITLE
focal loss calculation result is different?

### ISSUE THREAD

**Issue by yzl96** [open-mmlab/mmdetection#1299]
Hi, I use your framework focal loss, in focal_loss.py:
py_sigmoid_focal_loss function and FocalLoss class 's output is different, and I calculate with hands, and I find the result of py_sigmoid_focal_loss function is right.
For example
inputs: torch.tensor([0.1, 0.5, 0.4, 0.3]).reshape(1, -1).cuda()
targets: torch.tensor([2]).cuda()
FocalLoss class's  result: 0.1569
py_sigmoid_focal_loss's result: 0.1673 

what's more, will same config, if I use py_sigmoid_focal_loss , the loss quickly go nan, and if use FocalLoss class, the loss will not go nan, and gradually drop, but when validate, the mAP is 0, and I also test the model with image, the box predicted is all wrong.

**Comment by yhcao6** [comment]
In FocalLoss class, label 0 represents background. So you should set targets to be
` targets: torch.tensor([3]).cuda()` which is equivalent to ` targets: torch.tensor([2]).cuda()` in py_sigmoid_focal_loss
